scripts/printReleaseNotesTemplate.ts

```diff
@@ -1,55 +1,553 @@
+}
+
+interface ChangelogOptions {
+  dryRun?: boolean
+  title?: string
+}
+
+interface ReleaseContext {
+  version: string
+  title: string
+  product: string
+  features: string[]
+  bugfixes: string[]
+  changelogOutput: string
+}
+
+interface PortableTextSpan {
+  _key: string
+  _type: 'span'
+  marks: string[]
+  text: string
+}
+
+interface PortableTextBlock {
+  _key: string
+  _type: 'block'
+  children: PortableTextSpan[]
+  markDefs: Array<{
+    _key: string
+    _type: string
+    href: string
+  }>
+  style: string
+  level?: number
+  listItem?: string
+}
+
+// Parse command-line arguments
+function parseArguments() {
+  return yargs(hideBin(process.argv))
+    .option('dryRun', {
+      type: 'boolean',
+      description: 'Show documents that would be created without actually creating them',
+      default: false,
+    })
+    .option('title', {
+      type: 'string',
+      description: 'Title for the release',
+      default: '',
+    })
+    .option('debug', {
+      type: 'boolean',
+      description: 'Show document structure before creating',
+      default: false,
+    })
+    .option('token', {
+      type: 'string',
+      description: 'Sanity auth token (alternative to SANITY_WEB_AUTH_TOKEN env var)',
+    })
+    .help().argv as ChangelogOptions & {debug?: boolean; token?: string}
+}
+
+// Extract commit data and categorize into features and bugfixes
+function extractCommitData(): {features: string[]; bugfixes: string[]; changelogOutput: string} {
+  // Get the latest tag
+  const latestTag = execa.sync('git describe --abbrev=0', {shell: true}).stdout.trim()
+
+  // Generate changelog from latest tag to next branch
+  const CHANGELOG_COMMAND = `git log --pretty=format:'%aN | %s | %h' --abbrev-commit --reverse ${latestTag}..origin/next`
+  const changelogOutput = execa.sync(CHANGELOG_COMMAND, {shell: true}).stdout
+
+  // Extract feature and bugfix commits for highlights
+  const commits = changelogOutput.split('\n').filter((line: string) => line.trim() !== '')
+
+  // Parse commits using conventional commit format
+  const features = []
+  const bugfixes = []
+
+  for (const commit of commits) {
+    const parts = commit.split(' | ')
+    if (parts.length < 2) continue
+
+    const message = parts[1]
+    // Check for conventional commit format: type(scope): message
+    const conventionalMatch = message.match(
+      /^(feat|fix|chore|docs|style|refactor|perf|test|build|ci|revert)(\([^)]*\))?:\s*(.*)$/i,
+    )
+
+    if (conventionalMatch) {
+      const type = conventionalMatch[1].toLowerCase()
+
+      if (type === 'feat') {
+        features.push(commit)
+      } else if (type === 'fix') {
+        bugfixes.push(commit)
+      }
+      // Other types (chore, docs, etc.) are ignored
+    } else {
+      // Fallback to basic keyword matching only if not a conventional commit
+      const lowerMessage = message.toLowerCase()
+      if (lowerMessage.includes('feat') || lowerMessage.includes('feature')) {
```

Comment:
this will falsely mark commits with "feather" and "defeat" in the message as features